src/Signer.php

```diff
@@ -16,14 +15,14 @@ public function getAlgorithmId(): string;
     /**
      * Creates a hash for the given payload
      *
-     * @throws InvalidArgumentException When given key is invalid.
+     * @throws InvalidArgument When given key is invalid.
```

Comment:
In this case, I believe we had many things under the same exception type:

1. Unable to read the file contents
1. Key isn't parseable by OpenSSL
1. Key isn't compatible (ECDSA vs RSA keys)
1. Error whilst creating the signature (OpenSSL)
1. Invalid ASN1 sequence for ECDSA signature

Reply:
Don't have proper suggestions atm, though (shaaame)